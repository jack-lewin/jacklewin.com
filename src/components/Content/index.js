import React from 'react';
import contentProps from '../../prop-types/components/content';

export const HTMLContent = ({ content, className }) => (
  // eslint-disable-next-line react/no-danger
  <div className={className} dangerouslySetInnerHTML={{ __html: content }} />
);

const Content = ({ content, className }) => <div className={className}>{content}</div>;

Content.propTypes = contentProps;

HTMLContent.propTypes = Content.propTypes;

export default Content;
