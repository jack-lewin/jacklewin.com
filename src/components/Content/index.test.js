import React from 'react';
import { shallow } from 'enzyme';
import Content, { HTMLContent } from '.';

describe('Content', () => {
  const props = {
    content: <p>my safe jsx</p>,
    className: 'someClassName',
  };

  it('should render as expected', () => {
    const element = shallow(<Content {...props} />);
    expect(element).toMatchSnapshot();
  });
});

describe('HTMLContent', () => {
  const props = {
    content: '<p>my dangerous html</p>',
    className: 'someClassName',
  };

  it('should render as expected', () => {
    const element = shallow(<HTMLContent {...props} />);
    expect(element).toMatchSnapshot();
  });
});
