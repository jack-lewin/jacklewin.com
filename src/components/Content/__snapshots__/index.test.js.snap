// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content
    className="someClassName"
    content={
      <p>
        my safe jsx
      </p>
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        my safe jsx
      </p>,
      "className": "someClassName",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "my safe jsx",
      },
      "ref": null,
      "rendered": "my safe jsx",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          my safe jsx
        </p>,
        "className": "someClassName",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "my safe jsx",
        },
        "ref": null,
        "rendered": "my safe jsx",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`HTMLContent should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HTMLContent
    className="someClassName"
    content="<p>my dangerous html</p>"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "someClassName",
      "dangerouslySetInnerHTML": Object {
        "__html": "<p>my dangerous html</p>",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "someClassName",
        "dangerouslySetInnerHTML": Object {
          "__html": "<p>my dangerous html</p>",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
